src/core/src/MatrixDynSize.cpp

```diff
@@ -76,37 +77,32 @@ MatrixDynSize& MatrixDynSize::operator=(const MatrixDynSize& other)
 {
     if (this == &other) return *this;
 
-    // Copy the new rows
+    // Copy the new rows and columns
     m_rows = other.m_rows;
     m_cols = other.m_cols;
+    unsigned requiredCapacity = m_rows * m_cols;
+
+    // if other is empty, return
+    //FIXME: This probably changed the behaviour.
+    // the capacity is never changed if the other matrix is zero
+    if (requiredCapacity == 0) return  *this;
```

Comment:
> Honestly I thought the contrary. It is usually better to do "early return" as it simplify code reading.

Ok, then it is a personal taste. Nevermind, leave it like that. 

> About the previous behaviour. I think that before, if other is empty, then the internal buffer is freed, while now the capacity is not changed (which is usually a desired feature)

In the previous version (see https://github.com/robotology/idyntree/pull/281/files/6e6492904cf5915d28414dfa379845dba0d6aec1#diff-c1bdcea78a4cb86a68ea5b6a1dfccccfL87) if `m_capacity` was different from zero and m_rows*m_cols was zero the if would have been skipped, so I think the behaviour is the same modulo the bug that you fixed. I think we can drop the FIXME comment.